### ISSUE TITLE
First Knative Service expected URL is different from the output

### ISSUE THREAD

**Issue by hbelmiro** [knative/docs#4547]
When following the steps from https://knative.dev/docs/getting-started/first-service/, the outputted URL for the service is different from the expected. And when we try to access that URL, it can't be resolved.

curl http://hello.default.example.com
curl: (6) Could not resolve host: hello.default.example.com

## Expected Behavior

````
Service hello created to latest revision 'hello-world' is available at URL:
http://hello.default.127.0.0.1.nip.io
````

## Actual Behavior

````
Service 'hello' created to latest revision 'hello-world' is available at URL:
http://hello.default.example.com
````

## Steps to Reproduce the Problem

1. Run:

````
kn service create hello \
--image gcr.io/knative-samples/helloworld-go \
--port 8080 \
--env TARGET=World \
--revision-name=world
````

## Additional Info
macOS Monterey 12.0.1

**Install information:**
minikube version: v1.23.0
kn version: v0.27.0

**Comment by abrennan89** [comment]
@psschwei @omerbensaadon @csantanapr can ya'll take a look at this please since it's part of the getting started?

**Comment by psschwei** [comment]
What version of the plugin are you using? This is often a problem with the `minikube tunnel` command, and there's been a couple of updates to the latest version of the plugin (v1.0.1) to address common issues.

**Comment by csantanapr** [comment]
@hbelmiro try version v1.0.1 of the in-quickstart plugin I hit the same problem because of not stopping and asking user to run minikube tunnel

**Comment by nak3** [comment]
Hmm... I'm not minikube user so I may be wrong :sweat_smile: But Knative on minkube configures the domain `127.0.0.1.nip.io` from `example.com` automatically?
I thought that https://knative.dev/docs/getting-started/first-service/ requires the domain change in `config-domain` like [Configure DNS](https://knative.dev/docs/install/serving/install-serving-with-yaml/#configure-dns). Otherwise, Knaive uses the default `example.com`.

**Comment by hbelmiro** [comment]
> What version of the plugin are you using? This is often a problem with the `minikube tunnel` command, and there's been a couple of updates to the latest version of the plugin (v1.0.1) to address common issues.

@psschwei @csantanapr 
I'm using quickstart v1.0.1 and kn 0.27.0. Both installed from homebrew.

@nak3
From the link you provided:
> To configure DNS for Knative, take the External IP or CNAME from setting up networking, and configure it with your DNS provider as follows:

There's no way to take the External IP or CNAME from setting up networking and configure DNS. None of those services are installed (Kourier, Ambassador, Contour or Istio). I ran the Getting started tutorial, so I didn't any network configuration.

**Comment by psschwei** [comment]
From the info you've provided, it sounds like `minikube tunnel` is in fact the issue.

On Linux systems, the quickstart plugin runs the `minikube tunnel` command automatically, which would be why you didn't encounter an issue on Fedora. 

On Mac/Windows systems, the user needs to run the tunnel command themselves... after the cluster comes up, the screen should show a message that says:

```
To finish setting up networking for minikube, run the following command in a separate terminal window:
    minikube tunnel --profile knative

Press the Enter key to continue
```

Did you get this message when doing the setup on Mac?

**Comment by hbelmiro** [comment]
> Did you get this message when doing the setup on Mac?

Yes. I ran the command in a separate window and pressed Enter in the original one. I can't remember when I closed the tunnel window, but I'm sure I waited as much as it was reasonable for me. I'm not very familiar with k8s, so I can remove all and do a fresh install if you think that helps.

**Comment by psschwei** [comment]
One thing I just thought of... we just fixed the profile info for the tunnel in [#185](https://github.com/knative-sandbox/kn-plugin-quickstart/pull/185), so there's a chance that the tunnel was referencing the wrong profile... (actually, I think that's probably pretty likely the culprit here).

Can you try again, and when you run the tunnel in the separate window use `minikube tunnel --profile knative` (instead of `--profile minikube-knative`)? Hopefully that solves the issue...

**Comment by hbelmiro** [comment]
> Can you try again, and when you run the tunnel in the separate window use `minikube tunnel --profile knative` (instead of `--profile minikube-knative`)? Hopefully that solves the issue...

@psschwei 

````
~ minikube tunnel --profile knative
🤷  Profile "knative" not found. Run "minikube profile list" to view all profiles.
👉  To start a cluster, run: "minikube start -p knative"
````

**Comment by psschwei** [comment]
Hmmm... what profiles exist? (`minikube profile list`) 
Did you run the tunnel command separately or as part of the quickstart?

**Comment by csantanapr** [comment]
I think there seems to be some confusion here.

Would you mind using the released versions
kn version `v1.0.0`
```
❯ kn version
Version:      v1.0.0
Build Date:   2021-11-02 19:27:09
Git Revision: a1cf5bf8
Supported APIs:
* Serving
  - serving.knative.dev/v1 (knative-serving v0.27.0)
* Eventing
  - sources.knative.dev/v1 (knative-eventing v0.27.0)
  - eventing.knative.dev/v1 (knative-eventing v0.27.0)
```
kn-quickstart version `v1.0.1`
```
❯ kn quickstart version
Version:      v1.0.1
Build Date:   2021-11-23 09:07:26
Git Revision: 73f4d8c
```

@nak3 
>Hmm... I'm not a minikube user, so that I may be wrong 😅 But Knative on minkube configures the domain 127.0.0.1.nip.io from example.com automatically?
I thought that https://knative.dev/docs/getting-started/first-service/ requires the domain change in config-domain like Configure DNS. Otherwise, Knaive uses the default example.com which cannot be resolved.

The `kn quickstart` command installs kourier, sets the net class, and configures `.sslip.io` code here https://github.com/knative-sandbox/kn-plugin-quickstart/blob/main/pkg/install/install.go#L84

The profile name is `minikube-knative` in version v1.0.1, in HEAD/main is `knative` and next version `v1.1.0` will be `knative`
TLDR;
minikube tunnel -s minikube-knative (v1.0.1)
minikube tunnel -s knative (v1.1.x, or HEAD/main)

Please watch this movie, and you need to run the tunnel command in another terminal and leave it running while using knative.

https://user-images.githubusercontent.com/1094878/145648246-1a689314-c65e-45d3-98e3-f83d7435b46c.mp4


The command needs to run after creating minikube VM, and before installing knative, the `kn quickstart minikube` will ask the user to run tunnel, and return and press any key to continue.

One problem that I hit like once a year is with network routes on macOS; make sure there are no issues in the output of the `minikube tunnel` if there are errors with `10.x`, then you need to delete the route with these commands that I have aliases
```
alias minikube_list_route="route -n get 10.96/12"
alias minikube_fix_route="sudo route delete -net 10.96/12"
```
run `sudo route delete -net 10.96/12` or adjust based on your local IP Address

I'm using Big Sur. I have not tried macOS Monterey. I will update this weekend and try it out.
If you still have issues, please DM me on Knative Slack, and we can do a web conference and help you out and look at what's going on in your setup.

**Comment by psschwei** [comment]
> The profile name is `minikube-knative` in version v1.0.1, in HEAD/main is `knative` and next version `v1.1.0` will be `knative` TLDR; minikube tunnel --profile minikube-knative (v1.0.1) minikube tunnel --profile knative (v1.1.x(not released yet), or HEAD/main)

That's right, I got my wires crossed... the profile should be `minikube-knative`. Sorry for any confusion!

**Comment by hbelmiro** [comment]
Yeah. I left the tunnel window open and it worked. Sorry for that.
It might be a good idea to put a note on the docs saying that the window must remain open throughout the process.

**Comment by amandahla** [comment]
Hi, I had the same doubt but my output was:

```
kn service create hello \
--image gcr.io/knative-samples/helloworld-go \
--port 8080 \
--env TARGET=World \
--revision-name=world
Creating service 'hello' in namespace 'default':

  0.025s The Configuration is still working to reflect the latest desired specification.
  0.063s The Route is still working to reflect the latest desired specification.
  0.098s Configuration "hello" is waiting for a Revision to become ready.
 34.766s ...
 34.820s Ingress has not yet been reconciled.
 34.848s Waiting for load balancer to be ready
 35.059s Ready to serve.

Service 'hello' created to latest revision 'hello-world' is available at URL:
http://hello.default.127.0.0.1.sslip.io
```

I'm using kind instead of minikube.

Wouldn't be right to change the docs from "http://hello.default.127.0.0.1.nip.io" to "http://hello.default.127.0.0.1.sslip.io"?

Since this [doc](https://knative.dev/docs/install/serving/install-serving-with-yaml/#configure-dns) tells that the magic DNS is "sslip.io"

https://github.com/knative/docs/issues/3612

**Comment by psschwei** [comment]
Yes, that should be updated in the docs. Thanks for catching it!

**Comment by csantanapr** [comment]
/close
fixed in #4611
